lib/SGN/Controller/AJAX/LabelDesigner.pm

```diff
@@ -934,58 +934,88 @@ sub get_data {
         }
     }
     elsif ($data_level eq "crosses") {
-        my $project;
-        my $cross_list_ids;
-        my %all_design;
+        my %cross_info;
         if ($data_type =~ m/Crossing Experiments/) {
-            $project = CXGN::Cross->new({ schema => $schema, trial_id => $id});
-        } elsif ($data_type =~ m/List/) {
-            $cross_list_ids = convert_stock_list($c, $schema, $id);
-            my ($crossing_experiment_id, $num_trials) = get_trial_from_stock_list($c, $schema, $cross_list_ids);
-            $project = CXGN::Cross->new({ schema => $schema, trial_id => $crossing_experiment_id});
-        }
+            my $project = CXGN::Cross->new({ schema => $schema, trial_id => $id});
+            my $result = $project->get_crosses_and_details_in_crossingtrial();
+            my @cross_data = @$result;
+            foreach my $cross (@cross_data){
+                my $cross_combination;
+                my $male_parent_name;
+                my $male_parent_id;
+
+                if (!$cross->[2] || $cross->[2] eq ''){
```

Comment:
The code might be more readable if you assign specific variables with clear names instead of using $cross->[2]... what do you think?